File: user/lib/models/payment_model.dart
class Payment {
  final String id;
  final String bookingId;
  final double amount;
  final String paymentMethod;
  final String status;
  final String? transactionId;
  final String? receiptUrl;

  Payment({
    required this.id,
    required this.bookingId,
    required this.amount,
    this.paymentMethod = "stripe",
    this.status = "pending",
    this.transactionId,
    this.receiptUrl,
  });

  factory Payment.fromJson(Map<String, dynamic> json) => Payment(
    id: json['_id'],
    bookingId: json['bookingId'] is Map
        ? json['bookingId']['_id']
        : json['bookingId'],
    amount: (json['amount'] as num).toDouble(),
    paymentMethod: json['paymentMethod'] ?? "stripe",
    status: json['status'] ?? "pending",
    transactionId: json['transactionId'],
    receiptUrl: json['receiptUrl'],
  );

  Map<String, dynamic> toJson() => {
    "_id": id,
    "bookingId": bookingId,
    "amount": amount,
    "paymentMethod": paymentMethod,
    "status": status,
    "transactionId": transactionId,
    "receiptUrl": receiptUrl,
  };
}

File: backend/models/payment.model.js
import mongoose from "mongoose";

const paymentSchema = new mongoose.Schema(
  {
    bookingId: { type: mongoose.Schema.Types.ObjectId, ref: "Booking", required: true },
    amount: { type: Number, required: true },
    paymentMethod: { type: String, enum: ["card", "paypal", "stripe"], default: "stripe" },
    status: { type: String, enum: ["paid", "failed", "pending"], default: "pending" },
    transactionId: { type: String },
    receiptUrl: { type: String, default: "" }
  },
  { timestamps: true }
);

const Payment = mongoose.model("Payment", paymentSchema);
export default Payment;